File: LordOfTheGames/app/src/main/java/com/example/lordofthegames/EntityDAO/UserDAO.java
package com.example.lordofthegames.EntityDAO;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import com.example.lordofthegames.db_entities.User;

import java.util.List;

@Dao
public interface UserDAO {
    @Insert
    void insert(User user);

    @Query("SELECT * FROM User")
    List<User> getAllUsers();
}

File: LordOfTheGames/app/src/main/java/com/example/lordofthegames/EntityDAO/UserDAO.kt
package com.example.lordofthegames.EntityDAO

import androidx.lifecycle.LiveData
import androidx.room.*
import com.example.lordofthegames.db_entities.User

@Dao
interface UserDAO {
    @Query("SELECT * FROM User")
    fun getItems(): LiveData<List<User?>?>

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insert(item: User)

    @Delete
    suspend fun delete(item: User)

    @Query("DELETE FROM User")
    suspend fun deleteAll()

}